src/client/datascience/jupyter/kernels/cellExecution.ts

```diff
@@ -296,8 +296,8 @@ export class CellExecution {
         if (this.isEmptyCodeCell) {
             this.task?.end({});
         } else if (success === 'success' || success === 'failed') {
-            this.lastRunDuration = this.stopWatch.elapsedTime;
-            this.task?.end({ duration: this.lastRunDuration, success: success === 'success' });
+            this.endTime = new Date().getTime();
```

Comment:
Peng was right. I did this endTime convert quickly and I'm not fully sure that it's handling the elapsed time correctly. Let me re-look quick.

Reply:
Actually I think this is ok. Stopwatch still used to send durations for telemetry, we start it right where we set start time and just get an endTime here now, should be legit.